src/main/java/seedu/address/model/person/Address.java

```diff
@@ -27,11 +26,15 @@
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address) throws IllegalValueException {
-        requireNonNull(address);
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+
+        if (address == null) {
+            this.value = ADDRESS_NOT_ASSIGNED;
```

Comment:
This value is not marked as valid in Address::isValidAddress. It works currently because your ADDRESS_VALIDATION_REGEX is set to match anything (see Phone::isValidPhone).

Reply:
ok